### DISCUSSION TITLE
convert H265 to H164.

### DISCUSSION THREAD

**Discussion by deusyv** [bluenviron/mediamtx#930] [Questions]
Hello,
I have  cam h265

 ffprobe "rtsp://192.168.105.119:554/user=admin_password=xxx_channel=1_stream=0&protocol=unicast"
ffprobe version 4.4.2-1 Copyright (c) 2007-2021 the FFmpeg developers
  built with gcc 11 (Debian 11.2.0-20)
  configuration: --prefix=/usr --extra-version=1 --toolchain=hardened --libdir=/usr/lib/x86_64-linux-gnu --incdir=/usr/include/x86_64-linux-gnu --arch=amd64 --enable-gpl --disable-stripping --enable-gnutls --enable-ladspa --enable-libaom --enable-libass --enable-libbluray --enable-libbs2b --enable-libcaca --enable-libcdio --enable-libcodec2 --enable-libdav1d --enable-libflite --enable-libfontconfig --enable-libfreetype --enable-libfribidi --enable-libgme --enable-libgsm --enable-libjack --enable-libmp3lame --enable-libmysofa --enable-libopenjpeg --enable-libopenmpt --enable-libopus --enable-libpulse --enable-librabbitmq --enable-librubberband --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libsrt --enable-libssh --enable-libtheora --enable-libtwolame --enable-libvidstab --enable-libvorbis --enable-libvpx --enable-libwebp --enable-libx265 --enable-libxml2 --enable-libxvid --enable-libzimg --enable-libzmq --enable-libzvbi --enable-lv2 --enable-omx --enable-openal --enable-opencl --enable-opengl --enable-sdl2 --disable-sndio --enable-pocketsphinx --enable-librsvg --enable-libmfx --enable-libdc1394 --enable-libdrm --enable-libiec61883 --enable-chromaprint --enable-frei0r --enable-libx264 --enable-shared
  libavutil      56. 70.100 / 56. 70.100
  libavcodec     58.134.100 / 58.134.100
  libavformat    58. 76.100 / 58. 76.100
  libavdevice    58. 13.100 / 58. 13.100
  libavfilter     7.110.100 /  7.110.100
  libswscale      5.  9.100 /  5.  9.100
  libswresample   3.  9.100 /  3.  9.100
  libpostproc    55.  9.100 / 55.  9.100
Input #0, rtsp, from 'rtsp://192.168.105.119:554/user=admin_password=xxx_channel=1_stream=0&protocol=unicast':
  Metadata:
    title           : RTSP Session
  Duration: N/A, start: 0.066667, bitrate: N/A
  Stream #0:0: Video: hevc (Main), yuvj420p(pc), 1920x1080, 15 fps, 15 tbr, 90k tbn, 15 tbc

Frigate does not work directly in H265. Because Chrome or Firefox does not yet manage the H265

I want to convert The H265 to H164.
Camera ( RTSP Server) <-(RTSP Customer ) RTSP-Simple-Server (Pushisher) <-(RTSP customer) FRIGATE.

Can you indicate how to take it because I think badly and it doesn't work.

RTSP-Simple-Server config
...
paths:
  all:
    runOnReady: ffmpeg -i rtsp://192.168.105.119:554/user=admin_password=xxxx_channel=1_stream=0&protocol=unicast -pix_fmt yuv420p -c:v libx264 -preset ultrafast -b:v 600k -max_muxing_queue_size 1024 -f rtsp rtsp://localhost:$RTSP_PORT/compressed
    runOnReadyRestart: yes
 
 
 
 frigate config
 
 cameras:
  Camera1:
    ffmpeg:
      inputs:
        - path: rtsp://192.168.105.44:5541/compressed
        
        
        
rtsp-simple-server-1_1  |
rtsp-simple-server-1_1  |
rtsp-simple-server-1_1  | 2022/04/28 12:57:25 DEB [path compressed] created
rtsp-simple-server-1_1  | 2022/04/28 12:57:25 DEB [RTSP] [conn 192.168.105.42:38522] [s->c] RTSP/1.0 404 Not Found
rtsp-simple-server-1_1  | CSeq: 2
rtsp-simple-server-1_1  | Server: gortsplib
rtsp-simple-server-1_1  |
rtsp-simple-server-1_1  |
rtsp-simple-server-1_1  | 2022/04/28 12:57:25 DEB [path compressed] destroyed (not in use)
rtsp-simple-server-1_1  | 2022/04/28 12:57:25 INF [RTSP] [conn 192.168.105.42:38522] closed (no one is publishing to path 'compressed')
rtsp-simple-server-1_1  | 2022/04/28 12:57:29 INF [RTSP] [conn 192.168.105.2:1114] opened
rtsp-simple-server-1_1  | 2022/04/28 12:57:29 DEB [RTSP] [conn 192.168.105.2:1114] [c->s] OPTIONS rtsp://192.168.105.44:5541/compressed RTSP/1.0
rtsp-simple-server-1_1  | CSeq: 2
rtsp-simple-server-1_1  | User-Agent: LibVLC/3.0.16 (LIVE555 Streaming Media v2016.11.28)
rtsp-simple-server-1_1  |
rtsp-simple-server-1_1  |
rtsp-simple-server-1_1  | 2022/04/28 12:57:29 DEB [RTSP] [conn 192.168.105.2:1114] [s->c] RTSP/1.0 200 OK
rtsp-simple-server-1_1  | CSeq: 2
rtsp-simple-server-1_1  | Public: DESCRIBE, ANNOUNCE, SETUP, PLAY, RECORD, PAUSE, GET_PARAMETER, TEARDOWN
rtsp-simple-server-1_1  | Server: gortsplib
rtsp-simple-server-1_1  |
rtsp-simple-server-1_1  |
rtsp-simple-server-1_1  | 2022/04/28 12:57:29 DEB [RTSP] [conn 192.168.105.2:1114] [c->s] DESCRIBE rtsp://192.168.105.44:5541/compressed RTSP/1.0
rtsp-simple-server-1_1  | Accept: application/sdp
rtsp-simple-server-1_1  | CSeq: 3
rtsp-simple-server-1_1  | User-Agent: LibVLC/3.0.16 (LIVE555 Streaming Media v2016.11.28)
rtsp-simple-server-1_1  |
rtsp-simple-server-1_1  |
rtsp-simple-server-1_1  | 2022/04/28 12:57:29 DEB [path compressed] created
rtsp-simple-server-1_1  | 2022/04/28 12:57:29 DEB [RTSP] [conn 192.168.105.2:1114] [s->c] RTSP/1.0 404 Not Found
rtsp-simple-server-1_1  | CSeq: 3
rtsp-simple-server-1_1  | Server: gortsplib
rtsp-simple-server-1_1  |
rtsp-simple-server-1_1  |
rtsp-simple-server-1_1  | 2022/04/28 12:57:29 INF [RTSP] [conn 192.168.105.2:1114] closed (no one is publishing to path 'compressed')
rtsp-simple-server-1_1  | 2022/04/28 12:57:29 DEB [path compressed] destroyed (not in use)
rtsp-simple-server-1_1  | 2022/04/28 12:57:29 INF [RTSP] [conn 192.168.105.2:1115] opened
rtsp-simple-server-1_1  | 2022/04/28 12:57:29 DEB [RTSP] [conn 192.168.105.2:1115] [c->s] SETUP rtsp://192.168.105.44:5541/compressed RTSP/1.0
rtsp-simple-server-1_1  | CSeq: 0
rtsp-simple-server-1_1  | Transport: RTP/AVP;unicast;client_port=9110-9111
rtsp-simple-server-1_1  |
rtsp-simple-server-1_1  |
rtsp-simple-server-1_1  | 2022/04/28 12:57:29 INF [RTSP] [session 943000537] created by 192.168.105.2:1115
rtsp-simple-server-1_1  | 2022/04/28 12:57:29 DEB [RTSP] [conn 192.168.105.2:1115] [s->c] RTSP/1.0 400 Bad Request
rtsp-simple-server-1_1  | CSeq: 0
rtsp-simple-server-1_1  | Server: gortsplib
rtsp-simple-server-1_1  |
rtsp-simple-server-1_1  |
rtsp-simple-server-1_1  | 2022/04/28 12:57:29 INF [RTSP] [conn 192.168.105.2:1115] closed (path of a SETUP request must end with a slash. This typically happens when VLC fails a request, and then switches to an unsupported RTSP dialect)
rtsp-simple-server-1_1  | 2022/04/28 12:57:29 INF [RTSP] [conn 192.168.105.2:1116] opened
rtsp-simple-server-1_1  | 2022/04/28 12:57:29 DEB [RTSP] [conn 192.168.105.2:1116] [c->s] OPTIONS rtsp://192.168.105.44:5541 RTSP/1.0

can you help me?

Regard,

**Comment by deusyv** [comment]
Hello,
i change my config but
```
paths:
  all:
 
    
  compressed:
    runOnInit: ffmpeg -i rtsp://localhost:$RTSP_PORT/nl12_1 -map 0:v:0 -c:v libx264 -preset ultrafast -b:v 500k -max_muxing_queue_size 1024 -f rtsp rtsp://localhost:$RTSP_PORT/$RTSP_PATH
    runOnInitRestart: yes

  nl12_1:
    source: "rtsp://192.168.105.119:554/user=admin_password=xxx_channel=1_stream=0&protocol=unicast"
    sourceOnDemand: no
```

Starting transcode_rtsp-simple-server-1_1 ... done
Attaching to transcode_rtsp-simple-server-1_1
rtsp-simple-server-1_1  | 2022/04/28 13:53:41 INF rtsp-simple-server v0.18.1
rtsp-simple-server-1_1  | 2022/04/28 13:53:41 DEB [path compressed] created
rtsp-simple-server-1_1  | 2022/04/28 13:53:41 DEB [path nl12_1] created
rtsp-simple-server-1_1  | 2022/04/28 13:53:41 DEB path manager created
rtsp-simple-server-1_1  | 2022/04/28 13:53:41 INF [path nl12_1] [rtsp source] started
rtsp-simple-server-1_1  | 2022/04/28 13:53:41 DEB [path nl12_1] [rtsp source] connecting
rtsp-simple-server-1_1  | 2022/04/28 13:53:41 INF [path compressed] runOnInit command started
rtsp-simple-server-1_1  | 2022/04/28 13:53:41 INF [RTSP] listener opened on :8554 (TCP)
rtsp-simple-server-1_1  | 2022/04/28 13:53:41 INF [RTMP] listener opened on :1935
rtsp-simple-server-1_1  | 2022/04/28 13:53:41 INF [HLS] listener opened on :8888
rtsp-simple-server-1_1  | 2022/04/28 13:53:41 INF [path compressed] runOnInit command exited with code 0
rtsp-simple-server-1_1  | 2022/04/28 13:53:41 DEB [path nl12_1] [rtsp source] c->s OPTIONS rtsp://192.168.105.119:554/user=admin_password=xxxx_channel=1_stream=0&protocol=unicast RTSP/1.0
rtsp-simple-server-1_1  | CSeq: 1
rtsp-simple-server-1_1  | User-Agent: gortsplib
rtsp-simple-server-1_1  |
rtsp-simple-server-1_1  |
rtsp-simple-server-1_1  | 2022/04/28 13:53:41 DEB [path nl12_1] [rtsp source] s->c RTSP/1.0 200 OK
rtsp-simple-server-1_1  | CSeq: 1
rtsp-simple-server-1_1  | Public: OPTIONS, DESCRIBE, SETUP, TEARDOWN, GET_PARAMETER, SET_PARAMETER, PLAY, PAUSE
rtsp-simple-server-1_1  | Server: H264DVR 1.0
rtsp-simple-server-1_1  |
rtsp-simple-server-1_1  |
rtsp-simple-server-1_1  | 2022/04/28 13:53:41 DEB [path nl12_1] [rtsp source] c->s DESCRIBE rtsp://192.168.105.119:554/user=admin_password=xxxx_channel=1_stream=0&protocol=unicast RTSP/1.0
rtsp-simple-server-1_1  | Accept: application/sdp
rtsp-simple-server-1_1  | CSeq: 2
rtsp-simple-server-1_1  | User-Agent: gortsplib
rtsp-simple-server-1_1  |
rtsp-simple-server-1_1  |
rtsp-simple-server-1_1  | 2022/04/28 13:53:41 DEB [path nl12_1] [rtsp source] s->c RTSP/1.0 200 OK
rtsp-simple-server-1_1  | CSeq: 2
rtsp-simple-server-1_1  | Cache-Control: private
rtsp-simple-server-1_1  | Content-Base: rtsp://192.168.105.119:554/user=admin_password=xxx_channel=1_stream=0&protocol=unicast
rtsp-simple-server-1_1  | Content-Length: 392
rtsp-simple-server-1_1  | Content-Type: application/sdp
rtsp-simple-server-1_1  | Server: H264DVR 1.0
rtsp-simple-server-1_1  | X-Accept-Dynamic-Rate: 1
rtsp-simple-server-1_1  | X-Accept-Retransmit: our-retransmit
rtsp-simple-server-1_1  |
rtsp-simple-server-1_1  | v=0
rtsp-simple-server-1_1  | o=- 38990265062388 38990265062388 IN IP4 192.168.105.119
rtsp-simple-server-1_1  | s=RTSP Session
rtsp-simple-server-1_1  | t=0 0
rtsp-simple-server-1_1  | a=control:*
rtsp-simple-server-1_1  | a=range:npt=0-
rtsp-simple-server-1_1  | m=video 0 RTP/AVP 98
rtsp-simple-server-1_1  | c=IN IP4 192.168.105.119
rtsp-simple-server-1_1  | a=rtpmap:98 H265/90000
rtsp-simple-server-1_1  | a=fmtp:98 profile-id=010101;sprop-pps=RAHAcvAbJA==;sprop-sps=QgEBAWAAAAMAAAMAAAMAAAMAlqADwIARB8uKrTuiS7lZYQiwIANu6AAzf5gB;sprop-vps=QAEMAf//AWAAAAMAAAMAAAMAAAMAlqwJ;
rtsp-simple-server-1_1  | a=framerate:25
rtsp-simple-server-1_1  | a=control:trackID=3
rtsp-simple-server-1_1  |
rtsp-simple-server-1_1  | 2022/04/28 13:53:41 DEB [path nl12_1] [rtsp source] c->s SETUP rtsp://192.168.105.119:554/user=admin_password=xxx_channel=1_stream=0&protocol=unicast/trackID=3 RTSP/1.0
rtsp-simple-server-1_1  | CSeq: 3
rtsp-simple-server-1_1  | Transport: RTP/AVP;unicast;client_port=14256-14257;mode=play
rtsp-simple-server-1_1  | User-Agent: gortsplib
rtsp-simple-server-1_1  |
rtsp-simple-server-1_1  |
rtsp-simple-server-1_1  | 2022/04/28 13:53:41 DEB [path nl12_1] [rtsp source] s->c RTSP/1.0 200 OK
rtsp-simple-server-1_1  | CSeq: 3
rtsp-simple-server-1_1  | Cache-Control: private
rtsp-simple-server-1_1  | Server: H264DVR 1.0
rtsp-simple-server-1_1  | Session: 30663880;timeout=60
rtsp-simple-server-1_1  | Transport: RTP/AVP;unicast;mode=PLAY;source=192.168.105.119;client_port=14256-14257;server_port=40002-40003;ssrc=1
rtsp-simple-server-1_1  | X-Dynamic-Rate: 1
rtsp-simple-server-1_1  |
rtsp-simple-server-1_1  |
rtsp-simple-server-1_1  | 2022/04/28 13:53:41 INF [path nl12_1] [rtsp source] ready
rtsp-simple-server-1_1  | 2022/04/28 13:53:41 DEB [path nl12_1] [rtsp source] c->s PLAY rtsp://192.168.105.119:554/user=admin_password=xxx_channel=1_stream=0&protocol=unicast RTSP/1.0
rtsp-simple-server-1_1  | CSeq: 4
rtsp-simple-server-1_1  | Range: npt=0-
rtsp-simple-server-1_1  | Session: 30663880
rtsp-simple-server-1_1  | User-Agent: gortsplib
rtsp-simple-server-1_1  |
rtsp-simple-server-1_1  |
rtsp-simple-server-1_1  | 2022/04/28 13:53:41 DEB [path nl12_1] [rtsp source] s->c RTSP/1.0 200 OK
rtsp-simple-server-1_1  | CSeq: 4
rtsp-simple-server-1_1  | RTP-Info: url=trackID=3;seq=1;rtptime=0
rtsp-simple-server-1_1  | Range: npt=now-
rtsp-simple-server-1_1  | Server: H264DVR 1.0
rtsp-simple-server-1_1  | Session: 30663880
rtsp-simple-server-1_1  |
rtsp-simple-server-1_1  |
rtsp-simple-server-1_1  | 2022/04/28 13:53:46 INF [path compressed] runOnInit command exited with code 0
rtsp-simple-server-1_1  | 2022/04/28 13:53:49 INF [RTSP] [conn 192.168.105.42:39242] opened
rtsp-simple-server-1_1  | 2022/04/28 13:53:49 DEB [RTSP] [conn 192.168.105.42:39242] [c->s] OPTIONS rtsp://192.168.105.44:5541/compressed RTSP/1.0
rtsp-simple-server-1_1  | CSeq: 1
rtsp-simple-server-1_1  | User-Agent: Lavf58.45.100
rtsp-simple-server-1_1  |
rtsp-simple-server-1_1  |
rtsp-simple-server-1_1  | 2022/04/28 13:53:49 DEB [RTSP] [conn 192.168.105.42:39242] [s->c] RTSP/1.0 200 OK
rtsp-simple-server-1_1  | CSeq: 1
rtsp-simple-server-1_1  | Public: DESCRIBE, ANNOUNCE, SETUP, PLAY, RECORD, PAUSE, GET_PARAMETER, TEARDOWN
rtsp-simple-server-1_1  | Server: gortsplib
rtsp-simple-server-1_1  |
rtsp-simple-server-1_1  |
rtsp-simple-server-1_1  | 2022/04/28 13:53:49 DEB [RTSP] [conn 192.168.105.42:39242] [c->s] DESCRIBE rtsp://192.168.105.44:5541/compressed RTSP/1.0
rtsp-simple-server-1_1  | Accept: application/sdp
rtsp-simple-server-1_1  | CSeq: 2
rtsp-simple-server-1_1  | User-Agent: Lavf58.45.100
rtsp-simple-server-1_1  |
rtsp-simple-server-1_1  |
rtsp-simple-server-1_1  | 2022/04/28 13:53:49 DEB [RTSP] [conn 192.168.105.42:39242] [s->c] RTSP/1.0 404 Not Found
rtsp-simple-server-1_1  | CSeq: 2
rtsp-simple-server-1_1  | Server: gortsplib
rtsp-simple-server-1_1  |
rtsp-simple-server-1_1  |
rtsp-simple-server-1_1  | 2022/04/28 13:53:49 INF [RTSP] [conn 192.168.105.42:39242] closed (no one is publishing to path 'compressed')
rtsp-simple-server-1_1  | 2022/04/28 13:53:51 INF [path compressed] runOnInit command exited with code 0
rtsp-simple-server-1_1  | 2022/04/28 13:53:56 INF [path compressed] runOnInit command exited with code 0



RSTP-Simple Server listens to the cam but frigate or VLC still does not work

**Reply by aler9** [reply]
You have to start ffmpeg after the RTSP source has been correctly set-up:

```yml
paths:
  compressed:

  nl12_1:
    source: "rtsp://192.168.105.119:554/user=admin_password=xxx_channel=1_stream=0&protocol=unicast"
    sourceOnDemand: no
    runOnReady: ffmpeg -i rtsp://localhost:$RTSP_PORT/nl12_1 -map 0:v:0 -c:v libx264 -preset ultrafast -b:v 500k -max_muxing_queue_size 1024 -f rtsp rtsp://localhost:$RTSP_PORT/compressed
    runOnReadyRestart: yes
```

Furthermore, make sure that ffmpeg is correctly installed on the system. If you're using the docker image, you have to add ffmpeg to the image by following instructions [here](https://github.com/aler9/rtsp-simple-server/discussions/278#discussioncomment-549104).

[a comment by an automated account is not shown]
